### PR TITLE
Gen changes

### PR DESCRIPTION
**Pull request by galshi**

### CODE REVIEW

File: .s2i/bin/assemble [hunk 1]
```diff
@@ -23,6 +23,9 @@ npm install -g configurable-http-proxy
 cp /tmp/src/jupyterhub_config.py /opt/app-root/etc/
 cp /tmp/src/jupyterhub_config.sh /opt/app-root/etc/
 
+# cp /tmp/src/additional-jupyterhub_config.py /opt/app-root/etc/
```
Comment (galshi) [inline .s2i/bin/assemble]: You've created those files, so why don't you copy them?

File: .s2i/bin/assemble [hunk 2]
```diff
@@ -44,6 +47,8 @@ mv /tmp/src/scripts/* /opt/app-root/bin/
 
 mv /tmp/src/start-jupyterhub.sh /opt/app-root/bin/
 
+mv /tmp/src/config.yaml /opt/app-root/config.json
```
Comment (galshi) [inline .s2i/bin/assemble]: What is this config file?

File: Dockerfile [hunk 3]
```diff
@@ -22,4 +22,8 @@ ENV NPM_CONFIG_PREFIX=/opt/app-root \
 
 RUN /tmp/scripts/assemble
 
+# Enable jupyter_server_proxy
```
Comment (galshi) [inline Dockerfile]: What is this and why is it here if its disabled?

File: additional-jupyterhub_config.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
+def get_ldap_groups(username):
```
Comment (galshi) [inline additional-jupyterhub_config.py]: Why is this in a file if you also put it in a config map in the template?

File: config.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
+{
```
Comment (galshi) [inline config.yaml]: This is a json, not a yaml.
Also, you are passing this through a config map, so why is this here?

File: configMap-template.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+apiVersion: v1
```
Comment (galshi) [inline configMap-template.yaml]: What is this?

File: jupyterhub_config-workspace.py [hunk 7]
```diff
@@ -1,6 +1,12 @@
 # Authenticate users against OpenShift OAuth provider.
 
+from os import read
```
Comment (galshi) [inline jupyterhub_config-workspace.py]: Why do you need this import?

File: jupyterhub_config-workspace.py [hunk 8]
```diff
@@ -66,33 +75,7 @@
...
-                    "memory": os.environ.get('NOTEBOOK_MEMORY', '128Mi')
-                }
-            },
-            'volumeMounts': [
-                {
-                    'name': 'data',
-                    'mountPath': '/mnt'
-                }
-            ]
+            'subPath': 'private-workspace'
```
Comment (galshi) [inline jupyterhub_config-workspace.py]: Why is this private workspace?
Btw the pvc was never created in the Azure prod.

File: jupyterhub_config.py [hunk 9]
```diff
@@ -275,3 +272,12 @@ def _wrapper_get_env(wrapped, instance, args, kwargs):
 if os.path.exists(environ_config_file):
     with open(environ_config_file) as fp:
         exec(compile(fp.read(), environ_config_file, 'exec'), globals())
+
+
+# Load configuration provided via a config map.
+
+config_map_file = '/opt/app-root/etc/additional-jupyterhub_config.py'
```
Comment (galshi) [inline jupyterhub_config.py]: You don't need this.

File: jupyterhub_config.sh [hunk 10]
```diff
@@ -11,3 +11,8 @@ fi
 if [ -f /opt/app-root/configs/jupyterhub_config.sh ]; then
     . /opt/app-root/configs/jupyterhub_config.sh
 fi
+
+
+if [ -f /opt/app-root/etc/additional-jupyterhub_config.sh ]; then
```
Comment (galshi) [inline jupyterhub_config.sh]: You don't use this.

File: requirements - Copy.txt [hunk 11]
```diff
@@ -0,0 +1,11 @@
+kubernetes==11.0.0
```
Comment (galshi) [inline requirements - Copy.txt]: Why does this exist?

File: templates/jupyterhub-workspace.yaml [hunk 12]
```diff
@@ -0,0 +1,307 @@
+kind: Template
```
Comment (galshi) [inline templates/jupyterhub-workspace.yaml]: What is this?

File: templates/jupyterhub-deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,471 @@
...
+  value: test
+  description: LDAP_PASSWORD, in base64
+  required: true
+- name: LDAP_USERNAME
+  value: test
+  description: LDAP_USERNAME, in base64
+  required: true
+
+objects:
+#- apiVersion: v1
```
Comment (galshi) [inline templates/jupyterhub-deployment.yaml]: Pass this through the env variable interface `JUPYTERHUB_CONFIG`.

File: templates/jupyterhub-deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,471 @@
...
+            valueFrom:
+              secretKeyRef:
+                name: ldap
+                key: LDAP_USERNAME
+
+
+          volumeMounts:
+          - name: config
+            mountPath: "/opt/app-root/configs"
+#          - name: py-config
```
Comment (galshi) [inline templates/jupyterhub-deployment.yaml]: Get rid of this.

File: templates/jupyterhub-deployment.yaml [hunk 15]
```diff
@@ -0,0 +1,471 @@
...
+#            mountPath: /opt/app-root/pyconfigs
+        volumes:
+        - name: config
+          configMap:
+            name: "${APPLICATION_NAME}-cfg"
+            defaultMode: 420
+#        - name: py-config
```
Comment (galshi) [inline templates/jupyterhub-deployment.yaml]: Get rid of this.

File: additional-jupyterhub_config.py (hunk 4 above)
Comment (momoadc) [reply additional-jupyterhub_config.py]: that's where I originally developed it, it doesn't use it

File: config.yaml (hunk 5 above)
Comment (momoadc) [reply config.yaml]: that's where I originally developed it, it doesn't use it anymore

File: configMap-template.yaml (hunk 6 above)
Comment (momoadc) [reply configMap-template.yaml]: no  idea

File: requirements - Copy.txt (hunk 11 above)
Comment (momoadc) [reply requirements - Copy.txt]: again, used during development, it needed to be deleted

File: templates/jupyterhub-workspace.yaml (hunk 12 above)
Comment (momoadc) [reply templates/jupyterhub-workspace.yaml]: the old deployment template, I've created a new one to suit us better
